modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java

```diff
@@ -2993,6 +2994,64 @@ public void testDeleteObjectEntry() throws Exception {
 			publishedObjectDefinition);
 	}
 
+	@Test
+	public void testDeleteObjectEntryWithObjectDefinitionTree()
+		throws Exception {
+
+		Tree objectDefinitionTree = TreeTestUtil.createObjectDefinitionTree(
+			_objectDefinitionLocalService, _objectRelationshipLocalService,
+			true,
+			LinkedHashMapBuilder.put(
+				"A", new String[] {"AA", "AB"}
+			).put(
+				"AA", new String[] {"AAA", "AAB"}
+			).put(
+				"AB", new String[0]
+			).put(
+				"AAA", new String[0]
+			).put(
+				"AAB", new String[0]
+			).build());
+
+		ObjectDefinition rootObjectDefinition =
+			_objectDefinitionLocalService.getObjectDefinition(
+				objectDefinitionTree.getRootNode(
+				).getPrimaryKey());
+
+		Tree objectEntryTree = TreeTestUtil.createObjectEntryTree(
+			"1", _objectDefinitionLocalService, _objectEntryLocalService,
+			_objectFieldLocalService, _objectRelationshipLocalService,
+			rootObjectDefinition.getObjectDefinitionId());
+
+		ObjectEntry rootObjectEntry = _objectEntryLocalService.getObjectEntry(
+			objectEntryTree.getRootNode(
+			).getPrimaryKey());
```

Comment:
@victorKammerer let's avoid chaining (see this [article](https://liferay.atlassian.net/wiki/spaces/ENGR/pages/1511752519/Java+conventions+in+Liferay#%5BinlineExtension%5DAvoid-chaining)). You can change it to:

```java
ObjectEntry rootObjectEntry = _objectEntryLocalService.getObjectEntry(
	"A1", rootObjectDefinition.getObjectDefinitionId());
```